.github/actions/asdf-install-tooling/action.yml

```diff
@@ -109,25 +126,31 @@ runs:
           run: |
               asdf current
 
-              if [ -f .tool-versions ]; then
-                echo "Installing asdf plugins"
-
-                for plugin in $(awk '!/^#/ && NF {print $1}' .tool-versions); do
-                  if ! asdf plugin list | grep -q "^${plugin}$"; then
-                    asdf plugin add "${plugin}"
-                  else
-                    echo "Plugin ${plugin} already added"
-                  fi
-                done
-
-                echo "Update all asdf plugins"
-                asdf plugin update --all
-
-                asdf install
-
-              else
-                echo "No .tool-versions file found, skipping plugin installation."
-              fi
+              IFS=',' read -r -a tool_versions_files <<< "${{ inputs.tool_versions_files }}"
+              for tool_file in "${tool_versions_files[@]}"; do
+                if [ -f "$tool_file" ]; then
+                  echo "Installing asdf plugins from ${tool_file}"
+
+                  # Loop through each plugin in the current .tool-versions file
+                  for plugin in $(awk '!/^#/ && NF {print $1}' "$tool_file"); do
+                    # Add plugin if it is not already installed
+                    if ! asdf plugin list | grep -q "^${plugin}$"; then
+                      echo "Adding plugin ${plugin}"
+                      asdf plugin add "${plugin}"
+                    else
+                      echo "Plugin ${plugin} already added"
+                    fi
+                  done
+
+                  echo "Updating all asdf plugins"
+                  asdf plugin update --all
+
+                  # Install the tools specified in the current .tool-versions file
+                  asdf install
```

Comment:
correct me if I'm wrong but afaik this is not gonna do what you want it to do.
asdf is directory aware.

Meaning if you're calling `asdf install` from the root of the repository, it will install what is in that file and nothing of those in the subfolders.

If you call `asdf install` from the subfolder where the other `.tool-versions` resides, it's only gonna install those of the subfolder and if it exists in the root of the user.

Reply:
I think my comment remains the `asdf install` is happening on the `root` of the repository, so any subdirectory would be ignored.